doc/source/data/batch_inference.rst

```diff
@@ -0,0 +1,752 @@
+
+            .. annotations::
+                <1> Use the constructor (``__init__``) to initialize your model.
+
+                <2> The ``__call__`` method runs inference on a batch of data.
+
+
+3. Getting Predictions with Ray Data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once you have your Ray Datastream ``ds`` and your predictor class, you can use
+``ds.map_batches(...)`` to get predictions.
+Mapping batches this way is the recommended way to run inference with Ray, but there are also other options.
+``map_batches`` takes your predictor class as an argument and allows you to specify
+``compute`` resources by defining a so-called ``ActorPoolStrategy``.
+In the example below, we use two CPUs to run inference in parallel and then print the results.
+We cover resource allocation in more detail in :ref:`the configuration section of this guide<batch_inference_config>`.
+
+.. tabs::
+
+    .. group-tab:: HuggingFace
+
+        .. literalinclude:: ./doc_code/hf_quick_start.py
+            :language: python
+            :start-after: __hf_quickstart_prediction_start__
+            :end-before: __hf_quickstart_prediction_end__
+
+    .. group-tab:: PyTorch
+
+        .. literalinclude:: ./doc_code/pytorch_quick_start.py
+            :language: python
+            :start-after: __pt_quickstart_prediction_start__
+            :end-before: __pt_quickstart_prediction_end__
+
+    .. group-tab:: TensorFlow
+
+        .. literalinclude:: ./doc_code/tf_quick_start.py
+            :language: python
+            :start-after: __tf_quickstart_prediction_start__
+            :end-before: __tf_quickstart_prediction_end__
+
+.. note::
+
+    You can use ``map_batches`` on functions, too.
+    This is mostly useful for quick transformations of your data that doesn't require
+    an ML model or other stateful objects.
+    To handle state, using classes like we did above is the recommended way.
+    In the dropdown below you find an example of mapping data with a simple Python
+    function.
+
+    .. dropdown:: Example using ``map_batches`` with functions
+
+        This example transforms example data using a simple Python function.
+        The ``map_function`` uses the fact that our ``data`` batches in this particular
+        example are Pandas dataframes.
+        Note that by using a map function instead of a class, we don't have to define
+        ``ActorPoolStrategy`` to specify compute resources.
+
+        .. literalinclude:: ./doc_code/batch_formats.py
+            :language: python
+            :start-after: __simple_map_function_start__
+            :end-before: __simple_map_function_end__
+
+
+.. seealso::
+
+    As we've discussed in this guide, using ``map_batches`` on a class defining your model
+    should be your default choice for running inference with Ray.
+    However, there are other options, too.
+    For instance, if you're already using the Ray AIR framework for running your ML workflows,
```

Comment:
I believe the plan is to standardize on `map_batches` instead of the BatchPredictor interface going forward so there is only one way to do things, so I would remove this reference here :)

Reply:
cc @ericl

Reply:
agreed. let's leave it in for now and deprecate in the next round. this is not super urgent, I think. for now I consider this useful.